extensions/amp-story/1.0/media-pool.js

```diff
@@ -180,79 +168,40 @@ export class MediaPool {
     /** @private {?Array<!AmpElement>} */
     this.ampElementsToBless_ = null;
 
-    /** @const {!Object<string, (function(): !PoolBoundElementDef)>} */
-    this.mediaFactory_ = {
-      [MediaType.AUDIO]: () => {
-        const audioEl = this.win_.document.createElement('audio');
-        audioEl.setAttribute('muted', '');
-        audioEl.muted = true;
-        audioEl.classList.add('i-amphtml-pool-media');
-        audioEl.classList.add('i-amphtml-pool-audio');
-        return audioEl;
-      },
-      [MediaType.VIDEO]: () => {
-        const videoEl = this.win_.document.createElement('video');
-        videoEl.setAttribute('muted', '');
-        videoEl.muted = true;
-        videoEl.setAttribute('playsinline', '');
-        videoEl.classList.add('i-amphtml-pool-media');
-        videoEl.classList.add('i-amphtml-pool-video');
-        return videoEl;
-      },
-    };
-
-    this.initializeMediaPool_(maxCounts);
+    this.initializeMediaPool_(counts);
   }
 
   /**
    * Fills the media pool by creating the maximum number of media elements for
    * each of the types of media elements.  We need to create these eagerly so
    * that all media elements exist by the time that blessAll() is invoked,
    * thereby "blessing" all media elements for playback without user gesture.
-   * @param {!Object<!MediaType, number>} maxCounts The maximum amount of each
-   *     media element that can be allocated by the pool.
+   * @param {!Object<!MediaType, number>} counts
    * @private
    */
-  initializeMediaPool_(maxCounts) {
+  initializeMediaPool_(counts) {
     let poolIdCounter = 0;
-
-    this.forEachMediaType_((key) => {
-      const type = MediaType[key];
-      const count = maxCounts[type] || 0;
-
-      if (count <= 0) {
-        return;
-      }
-
-      const ctor = devAssert(
-        this.mediaFactory_[type],
-        `Factory for media type \`${type}\` unset.`
-      );
-
-      // Cloning nodes is faster than building them.
-      // Construct a seed media element as a small optimization.
-      const mediaElSeed = ctor.call(this);
```

Comment:
I feel a bit silly for having introduced these optimizations.

In practice, we only create up to 12 elements so it's marginal to optimize the loop, and cloning.